.github/workflows/main.yml

```diff
@@ -0,0 +1,85 @@
+name: Nar
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build-macos:
+    name: Build on macos
+    runs-on: macos-11
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install build dependencies
+      run: |
+        brew update
+        brew install automake
+    - uses: actions/setup-java@v2
+      with:
+        distribution: 'temurin'
+        java-version: '8'
+        cache: 'maven'
+    - name: Build and test the package
+      run: mvn test
+    - name: Store the build arctifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: macos-build
+        path: target/nar/louis-*
+
+  build-windows:
+    name: Build for Windows
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v2
+      with:
+        distribution: 'temurin'
+        java-version: '8'
+        cache: 'maven'
+    - name: Install build dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install texinfo libc6-dev-i386 mingw-w64
+    - name: Build and test the package
+      run: mvn test -Pcross-compile -Dhost.os=w64-mingw32 -Dos.arch=x86_64
+    - name: Store the build arctifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: windows-build
+        path: target/nar/louis-*
+
+  nar:
+    name: Build for linux and package the nar
+    needs: [build-macos, build-windows]
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v2
+      with:
+        distribution: 'temurin'
+        java-version: '8'
+        cache: 'maven'
+        server-id: sonatype-nexus-snapshots # sonatype-nexus-staging
```

Comment:
The server ID depends on whether we're building a snapshot or a release (which depends on the version number in the POM).

Reply:
I guess you really only want this action to run when there is a release

Reply:
That's a possibility. But it could also be useful to deploy snapshots.

In any case, the version number will need to be checked somehow. Or you could maybe only trigger the build when a tag is pushed.

Reply:
I'd use separate workflows for snapshots and releases that are triggered differently

Reply:
Isn't that a lot of duplication?